script.js

```diff
@@ -0,0 +1,178 @@
+  var middleLeft = document.getElementById("middle_left");
+  middleLeft.addEventListener("click", addImage);
+  var middleCenter = document.getElementById("middle_center");
+  middleCenter.addEventListener("click", addImage);
+  var middleRight = document.getElementById("middle_right");
+  middleRight.addEventListener("click", addImage);
+  var bottomLeft = document.getElementById("bottom_left");
+  bottomLeft.addEventListener("click", addImage);
+  var bottomCenter = document.getElementById("bottom_center");
+  bottomCenter.addEventListener("click", addImage);
+  var bottomRight = document.getElementById("bottom_right");
+  bottomRight.addEventListener("click", addImage);
+  var playAgain = document.getElementById("play_again");
+  playAgain.addEventListener("click", reset);
+}
+
+function updateTurn(player_current, player_other){
+  player_current.turn = false;
+  player_other.turn = true;
+  document.getElementById("winner_banner").innerText = player_other.name + "\'s turn";
+}
+
+function addImage(event) {
+  if (play){
+    var image = document.createElement("img");
+    this.appendChild(image);
+    this.removeEventListener("click", addImage);
+    if(player1.turn === true){
+      image.src = player1.token;
+      updateTttArray(this.id, player1);
+      updateTurn(player1, player2);
+      play = checkForWinner();
+    }
+    else if(player2.turn === true) {
+      image.src = player2.token;
+      updateTttArray(this.id, player2);
+      updateTurn(player2, player1);
+      play = checkForWinner();
+    }
+  }
+}
+
+function updateTttArray(id, player){
+  switch(id) {
+    case "top_left":
+      ttt_array[0][0] = player;
+      break;
+    case "top_center":
+      ttt_array[1][0] = player;
+      break;
+    case "top_right":
+      ttt_array[2][0] = player;
+      break;
+    case "middle_left":
+      ttt_array[0][1] = player;
+      break;
+    case "middle_center":
+      ttt_array[1][1] = player;
+      break;
+    case "middle_right":
+      ttt_array[2][1] = player;
+      break;
+    case "bottom_left":
+      ttt_array[0][2] = player;
+      break;
+    case "bottom_center":
+      ttt_array[1][2] = player;
+      break;
+    case "bottom_right":
+      ttt_array[2][2] = player;
+      break;
+  }
+
+}
+
+function lineNotNull(val1, val2, val3){
+  if(val1 == undefined || val2 == undefined || val3 == undefined){
+    return false;
+  }
+  return true;
+}
+
+function displayWinner(id1, id2, id3, player){
+  if(id1 === "draw"){
+      document.getElementById("winner_banner").innerText = "it's a draw";
+  }
+  else {
+    var invert_string = "invert_" + player.token;
+    document.getElementById("winner_banner").innerText = player.name + " wins!";
+    document.getElementById(id1).childNodes[0].src = invert_string;
+    document.getElementById(id2).childNodes[0].src = invert_string;
+    document.getElementById(id3).childNodes[0].src = invert_string;
+  }
+}
+
+function checkForWinner(){
+
+  turnCounter++;
```

Comment:
It might be beneficial to create a winConditions array and check against that instead of doing triple conditional expressions for each win condition